src/imagery/i.landsat/i.landsat.download/i.landsat.download.py

```diff
@@ -169,6 +176,119 @@ def normalize_time(datetime_str: str):
 
 
+def get_custom_eodag_config(datasource, settings):
+    username = None
+    password = None
+    apikey = None
+    if settings == "-":
+        # stdin
+        import getpass
+
+        if datasource == "usgs":
+            username = input(_("Insert username: "))
+            password = getpass.getpass(_("Insert password: "))
+        elif datasource == "planetary_computer":
+            apikey = getpass.getpass(_("Insert API key: "))
+    else:
+        try:
+            with open(settings, "r") as fd:
+                lines = list(
+                    filter(None, (line.rstrip() for line in fd))
+                )  # non-blank lines only
+                if datasource == "usgs":
+                    if len(lines) < 2:
+                        gs.fatal(_("Invalid settings file"))
+                    username = lines[0].strip()
+                    password = lines[1].strip()
+                elif datasource == "planetary_computer":
+                    if len(lines) < 1:
+                        gs.fatal(_("Invalid settings file"))
+                    apikey = lines[0].strip()
+        except IOError as e:
+            gs.fatal(_("Unable to open settings file: {}").format(e))
+
+    geojson_temp_dir = gs.tempdir()
+    custom_eodag_config_path = os.path.join(
+        geojson_temp_dir, f"{datasource}_config.yml"
+    )
+
+    eodag_config = None
+    try:
+        with open(os.path.expanduser("~/.config/eodag/eodag.yml")) as def_config:
+            eodag_config = yaml.safe_load(def_config)
+    except FileNotFoundError:
+        pass
+
+    def build_default_usgs_config(username, password):
+        usgs_config = {"priority": None, "api": {}}
+        usgs_config["api"]["extract"] = None
+        usgs_config["api"]["outputs_prefix"] = None
+        usgs_config["api"]["dl_url_params"] = None
+        usgs_config["api"]["product_location_scheme"] = None
+        usgs_config["api"]["credentials"] = {}
+        usgs_config["api"]["credentials"]["username"] = username
+        usgs_config["api"]["credentials"]["password"] = password
+        return usgs_config
+
+    def build_default_planetary_computer_config(apikey):
+        planetary_computer_config = {
+            "priority": None,
+            "search": None,
+            "auth": {},
+            "download": {},
+        }
+        planetary_computer_config["auth"]["credentials"] = {"apikey": apikey}
+        planetary_computer_config["download"]["outputs_prefix"] = None
+        return planetary_computer_config
+
+    if datasource == "usgs":
+        if username is None or password is None:
+            gs.warnign(_("No user or password given for USGS"))
+        else:
+            if eodag_config is None:
+                # Shall we always just ignore any configuration in the eodag config file
+                # to eliminate confusion between i.eodag and i.landsat.download?
+                eodag_config = {"usgs": build_default_usgs_config(username, password)}
```

Comment:
I do not like the idea of ignoring the eodag config file. 

Why not read the config file if available? Then, if credentials are passed via settings (file or interactive), maybe do what you suggest, i.e. create another config file for landsat? But will users end up with different eodag config files for different providers then? I'm thinking I search with usgs one day and another day with planetary, so I'll have my 2 text files and hidden there will be two config files one for usgs and one for planetary? or just one extra config file with both usgs and planetary? or will these be temporary? What do others think?

Reply:
> Why not read the config file if available?

What I was thinking is that it could be simpler to not have users worry about the eodag config file... for example, they may have changed any of the parameters values, e.g. delete_url_params, or any other parameter when they were working with i.eodag, so it could be confusing if then use i.landsat.download and see that the module's behaviour changed.

> But will users end up with different eodag config files for different providers then? I'm thinking I search with usgs one day and another day with planetary, so I'll have my 2 text files and hidden there will be two config files one for usgs and one for planetary? or just one extra config file with both usgs and planetary? or will these be temporary? 

No there won't be any extra config files for each provider... they will be temporary files that are created by the module automatically and filled with the credentials. So users won't have to create these config YAML files manually, instead they will just use the settings text file normally, and then i.landsat.download will create temporary file with the credentials and pass that temporary file to i.eodag.

Reply:
Ok, I see, in this way i.landsat.download will work as before (sort of)... Still users will need eodag and i.eodag installed. 

I would still keep the possibility to use the deafult eodag config file if it exists in the default location. Does that make sense? i.e., if I do not use the settings option, the default eodag config file should be used... is that too messy in terms of code? @ninsbl @lucadelu what do you think?

Reply:
>  is that too messy in terms of code? 

No it is not messy it will be a small change, it is almost what the code currently does. Currently, it creates a temporary copy of the default config file, if it exists, with the credentials added to it.

Reply:
> Ok, I see, in this way i.landsat.download will work as before (sort of)... Still users will need eodag and i.eodag installed.
> 
> I would still keep the possibility to use the deafult eodag config file if it exists in the default location. Does that make sense? i.e., if I do not use the settings option, the default eodag config file should be used... is that too messy in terms of code? @ninsbl @lucadelu what do you think? Also, should the same logic be applied to i.sentinel.download then?

I also like the idea to use the general eodag config, I understood that in this way for user it is a simpler change to use the new implementation, however could you check if eodag config file exists otherwise use the settings option?